Tests/tests/test_SystemServices.cpp

```diff
@@ -3700,7 +3700,7 @@ TEST_F(SystemServicesEventTest, onMacAddressesRetrieved)
             if (valueToReturn != NULL) {
                   char buffer[1024];
                   memset(buffer, 0, sizeof(buffer));
-                  strcpy(buffer, valueToReturn);
+                  strncpy(buffer, valueToReturn, sizeof(buffer) - 1);
```

Comment:
Checking why we are doing sizeof(buffer) - 1. It should be sizeof(buffer). it's already memset to 0. Pls check other strncpy where we have added -1

Reply:
So for 
`strncpy(buffer, valueToReturn, sizeof(buffer) - 1);`
the reason I do - 1 here is because the last byte in the buffer is already set to null because of the memset call, so I figured we never want to overwrite it so that the buffer is always null terminated no matter what.

The issue with `strncpy(dest, src, n);` 
If n is reached before the end of src, no null terminator is added. So for example
```
string str = "abcdefgh";
char buff[5];
memset(buff, '\0', sizeof(buff));
strncpy(buff, src, sizeof(buff)); 
```
Here the buffer would contain "abcde", not abcd\0.

If there wasn't a memset, then I would have to add the below line of code after to ensure there is always a null terminator
`buff[sizeof(buff) - 1] = '\0';`